hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieMultiTableDeltaStreamer.java

```diff
@@ -0,0 +1,269 @@
+      String schemaRegistrySuffix = typedProperties.getString(Constants.SCHEMA_REGISTRY_URL_SUFFIX_PROP);
+      typedProperties.setProperty(Constants.SOURCE_SCHEMA_REGISTRY_URL_PROP, schemaRegistryBaseUrl + typedProperties.getString(Constants.KAFKA_TOPIC_PROP) + schemaRegistrySuffix);
+      typedProperties.setProperty(Constants.TARGET_SCHEMA_REGISTRY_URL_PROP, schemaRegistryBaseUrl + typedProperties.getString(Constants.KAFKA_TOPIC_PROP) + schemaRegistrySuffix);
+    }
+  }
+
+  public static class Helpers {
+
+    static String getDefaultConfigFilePath(String configFolder, String database, String currentTable) {
+      return configFolder + Constants.FILEDELIMITER + database + Constants.UNDERSCORE + currentTable + Constants.DEFAULT_CONFIG_FILE_NAME_SUFFIX;
+    }
+
+    static String getTableWithDatabase(TableExecutionContext context) {
+      return context.getDatabase() + Constants.DELIMITER + context.getTableName();
+    }
+
+    static void deepCopyConfigs(Config globalConfig, Config tableConfig) {
+      tableConfig.enableHiveSync = globalConfig.enableHiveSync;
+      tableConfig.schemaProviderClassName = globalConfig.schemaProviderClassName;
+      tableConfig.sourceOrderingField = globalConfig.sourceOrderingField;
+      tableConfig.sourceClassName = globalConfig.sourceClassName;
+      tableConfig.tableType = globalConfig.tableType;
+      tableConfig.propsFilePath = globalConfig.propsFilePath;
+      tableConfig.basePathPrefix = globalConfig.basePathPrefix;
+      tableConfig.targetTableName = globalConfig.targetTableName;
+      tableConfig.configFolder = globalConfig.configFolder;
+      tableConfig.operation = globalConfig.operation;
+      tableConfig.sourceLimit = globalConfig.sourceLimit;
+      tableConfig.checkpoint = globalConfig.checkpoint;
+      tableConfig.continuousMode = globalConfig.continuousMode;
+      tableConfig.filterDupes = globalConfig.filterDupes;
+      tableConfig.payloadClassName = globalConfig.payloadClassName;
+      tableConfig.configs = globalConfig.configs;
+      tableConfig.forceDisableCompaction = globalConfig.forceDisableCompaction;
+      tableConfig.maxPendingCompactions = globalConfig.maxPendingCompactions;
+      tableConfig.minSyncIntervalSeconds = globalConfig.minSyncIntervalSeconds;
+      tableConfig.transformerClassName = globalConfig.transformerClassName;
+      tableConfig.commitOnErrors = globalConfig.commitOnErrors;
+      tableConfig.compactSchedulingMinShare = globalConfig.compactSchedulingMinShare;
+      tableConfig.compactSchedulingWeight = globalConfig.compactSchedulingWeight;
+      tableConfig.deltaSyncSchedulingMinShare = globalConfig.deltaSyncSchedulingMinShare;
+      tableConfig.deltaSyncSchedulingWeight = globalConfig.deltaSyncSchedulingWeight;
+      tableConfig.sparkMaster = globalConfig.sparkMaster;
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+    final Config config = new Config();
```

Comment:
Just to reiterate my comment related to configs, Can you create a new inner Config class for HoodieMultiTableDeltaStreamer with JCommander arguments to decouple from HoodieDeltaStreamer.Config. This should avoid keeping basePathPrefix in HoodieDeltaStreamer.Config

Reply:
Sure. Initially I was thinking of not creating the Config class in HoodieMultiTableDeltaStreamer as I thought it is going to be a duplicate of HoodieDeltaStreamer.Config class and hence leads to code duplication. But since you want to decouple the 2 classes, will do that in my next commit.